pkg/metrics/prometheus.go

```diff
@@ -235,3 +259,55 @@ func FromContext(ctx context.Context, name string) MetricCollector {
 	}
 	return NewPrometheus(name)
 }
+
+// statusRecorder to record the status code from the ResponseWriter
+type statusRecorder struct {
+	http.ResponseWriter
+	statusCode int
+}
+
+func (rec *statusRecorder) WriteHeader(statusCode int) {
+	rec.statusCode = statusCode
+	rec.ResponseWriter.WriteHeader(statusCode)
+}
+
+// MeasureGraphQLResponseDuration creates a middleware that records the response time and status code
+func (pc *prometheusCollector) MeasureGraphQLResponseDuration(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+
+		// Create a copy of the request body
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		r.Body = io.NopCloser(bytes.NewBuffer(body)) // Replace the request body with a copy
+
+		// Create another copy for JSON unmarshalling
+		bodyCopy := make([]byte, len(body))
+		copy(bodyCopy, body)
+
+		// Parse the operation name from the request body copy
+		var graphqlRequest struct {
+			OperationName string `json:"operationName"`
+		}
+		err = json.Unmarshal(bodyCopy, &graphqlRequest)
```

Comment:
nit: `if err := json.Unmarshal(bodyCopy, &graphqlRequest); err != nil {}`

Reply:
Took care of it. Thanks